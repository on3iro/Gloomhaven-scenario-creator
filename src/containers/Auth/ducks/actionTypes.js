export const LOGIN_SUBMIT = 'LOGIN_SUBMIT';
export const LOGIN_REQUEST = 'LOGIN_REQUEST';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGIN_ERROR = 'LOGIN_ERROR';
export const LOGOUT = 'LOGOUT';
