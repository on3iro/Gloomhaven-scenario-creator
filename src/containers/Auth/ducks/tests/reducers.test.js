import authReducer from '../reducers';
import * as types from '../actionTypes';


describe('authReducer', () => {
  it('should return initial state', () => {
    expect(authReducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle LOGIN_SUCCESS', () => {
    expect(authReducer(undefined, {
      type: types.LOGIN_SUCCESS,
      payload: {
        token: 'testtoken',
        user: 'testuser',
      }
    })).toMatchSnapshot();
  });

  it('should handle REGISTER_ERROR', () => {
    expect(authReducer(undefined, {
      type: types.REGISTER_ERROR,
      payload: 'register-error',
    })).toMatchSnapshot();
  });

  it('should handle LOGIN_ERROR', () => {
    expect(authReducer(undefined, {
      type: types.LOGIN_ERROR,
      payload: 'login-error',
    })).toMatchSnapshot();
  });

  it('should handle LOGOUT', () => {
    expect(authReducer({
      user: {
        isLoggedIn: true,
        token: 'testoken',
      },
    }, {
      type: types.LOGOUT,
    })).toMatchSnapshot();
  });
});
