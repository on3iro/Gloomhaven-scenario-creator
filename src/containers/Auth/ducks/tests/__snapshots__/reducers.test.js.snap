exports[`authReducer should handle LOGIN_ERROR 1`] = `
Object {
  "user": Object {
    "data": Object {},
    "error": "login-error",
    "isLoggedIn": false,
    "token": "",
  },
}
`;

exports[`authReducer should handle LOGIN_SUCCESS 1`] = `
Object {
  "user": Object {
    "data": "testuser",
    "error": "",
    "isLoggedIn": true,
    "token": "testtoken",
  },
}
`;

exports[`authReducer should handle LOGOUT 1`] = `
Object {
  "user": Object {
    "data": Object {},
    "isLoggedIn": false,
    "token": "",
  },
}
`;

exports[`authReducer should handle REGISTER_ERROR 1`] = `
Object {
  "user": Object {
    "data": Object {},
    "error": "register-error",
    "isLoggedIn": false,
    "token": "",
  },
}
`;

exports[`authReducer should return initial state 1`] = `
Object {
  "user": Object {
    "data": Object {},
    "error": Object {},
    "isLoggedIn": false,
    "token": "",
  },
}
`;
